File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int ans = INT_MAX;

        int i = 0,j = 0;
        int sz = nums.size();
        int sum = 0;
        while(j<sz) {
            sum+=nums[j];
            
            while(sum>=target && i<=j) {
                ans = min(ans,j-i+1);
                sum-=nums[i];
                i++;
            }
            j++;

        }
       

        if(ans == INT_MAX) ans = 0;
        return ans;
        
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int ans = Integer.MAX_VALUE;

        int n = nums.length;
        int j = 0;
        int i = 0;
        int sum = 0;
        while(j<n) {

            sum+=nums[j];
            while(sum>=target) {

                ans = Math.min(ans,j-i+1);
                sum-=nums[i];
                i++;
            }

            

            j++;
        }


        if(ans==Integer.MAX_VALUE) return 0;
        return ans;
    }
}